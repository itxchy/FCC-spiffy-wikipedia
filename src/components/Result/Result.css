.result-cell {
  min-height: 12rem;
  margin-bottom: .2rem;
  padding: .2rem 2rem 1.6rem;
  transition: ease .06s; }
  .result-cell:hover {
    background: #ecf4f9; }

.result-headline {
  text-align: center; }

.result-description {
  text-align: left;
  font-size: 1.2em; }
