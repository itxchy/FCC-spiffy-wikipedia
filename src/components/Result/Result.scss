.result-cell {
  min-height: 12rem;
  margin-bottom: .2rem;
  padding: .2rem 2rem 1.6rem;
  transition: ease .05s;
  &:hover {
    background: lighten(#d8e9f3, 5%);
  }
}