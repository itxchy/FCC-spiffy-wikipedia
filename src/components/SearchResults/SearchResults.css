.results {
  max-width: 80rem;
  margin: 2rem auto;
}

h1, h2 {
  font-family: 'Poiret One', cursive;
}
