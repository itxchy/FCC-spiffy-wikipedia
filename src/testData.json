[
  "cactus",
  [
    "Cactus",
    "Cactus Air Force",
    "Cactus (American band)",
    "Cactus Bowl",
    "Cactus Flower (film)",
    "Cactus, Texas",
    "Cactus (Indian band)",
    "Cactus wren",
    "Cactus Jack (band)",
    "Cactus Hill"
  ],
  [
    "A cactus (plural: cacti, cactuses, or cactus) is a member of the plant family Cactaceae, a family comprising about 127 genera with some 1750 known species of the order Caryophyllales.",
    "Cactus Air Force refers to the ensemble of Allied air power assigned to the island of Guadalcanal from August 1942 until December 1942 during the early stages of the Guadalcanal Campaign, particularly those operating from Henderson Field.",
    "Cactus is an American hard rock band, formed in 1969.",
    "The Cactus Bowl, officially the Motel 6 Cactus Bowl for sponsorship purposes, is an NCAA Football Bowl Subdivision college football bowl game that has been played in the state of Arizona since 1989.",
    "Cactus Flower is a 1969 American comedy film directed by Gene Saks and starring Walter Matthau, Ingrid Bergman, and Goldie Hawn, who won an Oscar for her performance.",
    "Cactus is a city in Moore County, Texas, United States, along U.S. Route 287. The population was 3,179 at the 2010 census.",
    "Cactus (Bengali: ক্যাকটাস) is an Indian Bengali rock band from Kolkata, West Bengal, India. Since inception they have recorded four full-length studio albums and have performed over 2500 concerts across India & abroad.",
    "The cactus wren (Campylorhynchus brunneicapillus) is a species of wren that is native to the southwestern United States southwards to central Mexico.",
    "Cactus Jack is a Serbian rock band from Pančevo.",
    "Cactus Hill is an archaeological site in southeastern Virginia, United States located on sand dunes above the Nottoway River about 45 miles south of Richmond."
  ],
  [
    "https://en.wikipedia.org/wiki/Cactus",
    "https://en.wikipedia.org/wiki/Cactus_Air_Force",
    "https://en.wikipedia.org/wiki/Cactus_(American_band)",
    "https://en.wikipedia.org/wiki/Cactus_Bowl",
    "https://en.wikipedia.org/wiki/Cactus_Flower_(film)",
    "https://en.wikipedia.org/wiki/Cactus,_Texas",
    "https://en.wikipedia.org/wiki/Cactus_(Indian_band)",
    "https://en.wikipedia.org/wiki/Cactus_wren",
    "https://en.wikipedia.org/wiki/Cactus_Jack_(band)",
    "https://en.wikipedia.org/wiki/Cactus_Hill"
  ]
]