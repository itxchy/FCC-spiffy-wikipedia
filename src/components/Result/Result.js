import React from 'react'
import './Result.css'
const { string } = React.PropTypes

function Result (props) {
  return (
    <article className="result-cell">
      <a href={this.props.link}>
        <h2>{this.props.headline}</h2>
      </a>
      <p>{this.props.description}</p>
    </article>
  )
}

Result.propTypes = {
  link: string,
  headline: string,
  description: string
}

export default Result
