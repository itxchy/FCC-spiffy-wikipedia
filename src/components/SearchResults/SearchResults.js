import React, { Component} from 'react'
import './SearchResults.css'
import Result from '../Result/Result'
const { array, object } = React.PropTypes

class SearchResults extends Component {
  constructor (props) {
    super(props)

    this.state = {
      populatedResults: false,
      noResults: null,
    }
  }

  errorCheck () {
    if (this.props.error) {
      return true
    }
    return false
  }

  render () {
    let populatedResultComponents = []
    const noResultsMessage = <h2>Nothing Found</h2>
    const errorMessage = <h2>An error occured.</h2>
    let nothingFound = false

    if (!this.props.error && this.props.searchResults.length > 1 && this.props.searchResults[1].length === 0) {
      console.log('this.props.searchResults', this.props.searchResults)
      nothingFound = true
      populatedResultComponents = false
    } 

    if (this.props.searchResults && this.props.searchResults[1]) {
      populatedResultComponents = this.props.searchResults[1].map((result, i) => {
        let element = (
          <Result
            headline={this.props.searchResults[1][i]}
            description={this.props.searchResults[2][i]}
            link={this.props.searchResults[3][i]}
            key={i}
          />
        )
        return element
      })
    }

    return (
      <section className='results'>
        {populatedResultComponents ? populatedResultComponents : null}
        {nothingFound ? noResultsMessage : null}
        {this.props.error ? errorMessage : null}
      </section>
    )
  }
}

SearchResults.propTypes = {
  searchResults: array,
  error: object
}

export default SearchResults
