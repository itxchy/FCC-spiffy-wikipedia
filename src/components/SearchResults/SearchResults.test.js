import React from 'react'
import { shallow, render } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'
import testData from '../../testData.json'
import SearchResults from './SearchResults'
import Result from '../Result/Result'

test('SearchResults snapshot test', () => {
  const component = shallow(<SearchResults error={null} searchResults={[]} />)
  const tree = shallowToJson(component)
  expect(tree).toMatchSnapshot()
})

test('SearchResults should render a Results component for each Result', () => {
  const component = shallow(<SearchResults error={null} searchResults={testData} />)
  expect(component.find(Result).length).toEqual(testData[1].length)
})

test('SearchResults should render "Nothing Found" if there are no results', () => {
  const component = shallow(<SearchResults error={null} searchResults={["nothing", [], [], [], []]} />)
  expect(component.text()).toEqual('Nothing Found')
})

test('SearchResults should render "An error occured." if an error is passed to props', () => {
  const component = shallow(<SearchResults error={{ error: 'bad news' }} searchResults={null} />)
  expect(component.text()).toEqual('An error occured.')
})