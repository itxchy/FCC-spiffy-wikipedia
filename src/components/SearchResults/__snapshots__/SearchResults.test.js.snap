exports[`test SearchResults snapshot test 1`] = `
<section
  className="results" />
`;
